src/test/java/it/gov/pagopa/reward/notification/service/rewards/evaluate/notify/RewardsNotificationExpiredInitiativeHandlerServiceIntegrationTest.java

```diff
@@ -0,0 +1,117 @@
+
+    private void storeTestData() {
+        // initiatives
+        expiredInitiative = ruleRepository.save(
+                RewardNotificationRuleFaker.mockInstanceBuilder(1)
+                        .initiativeId("INITIATIVE1")
+                        .endDate(TODAY.minusDays(5))
+                        .build()
+        ).block();
+        validInitiative = ruleRepository.save(
+                RewardNotificationRuleFaker.mockInstanceBuilder(2)
+                        .initiativeId("INITIATIVE2")
+                        .endDate(TODAY.plusDays(30))
+                        .build()
+        ).block();
+
+        // notifications
+        expiredNotification1 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(1)
+                        .initiativeId("INITIATIVE1")
+                        .notificationDate(TODAY.minusDays(6))
+                        .build()
+        ).block();
+        expiredNotification2 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(2)
+                        .initiativeId("INITIATIVE1")
+                        .notificationDate(null)
+                        .build()
+        ).block();
+        validNotification1 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(3)
+                        .initiativeId("INITIATIVE2")
+                        .notificationDate(TODAY)
+                        .build()
+        ).block();
+        validNotification2 = notificationRepository.save(
+                RewardsNotificationFaker.mockInstanceBuilder(4)
+                        .initiativeId("INITIATIVE2")
+                        .notificationDate(null)
+                        .build()
+        ).block();
+    }
+
+    @BeforeEach
+    void init() {
+        service = new RewardsNotificationExpiredInitiativeHandlerServiceImpl(ruleRepository, notificationRepository, dayBefore);
+        storeTestData();
+    }
+
+    @AfterEach
+    void clearData() {
+        ruleRepository.deleteAll().block();
+        notificationRepository.deleteAll().block();
+    }
+
+    @Test
+    void testHandle() {
+
+        // When
+        List<RewardsNotification> result = service.handle().collectList().block();
+
+        // Then
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(1, result.size());
+        checkHandleResult(expiredNotification2, result.get(0));
+    }
+
+    void checkHandleResult(RewardsNotification expected, RewardsNotification actual) {
+        Assertions.assertEquals(expected.getId(), actual.getId());
+        Assertions.assertEquals(expected.getInitiativeId(), actual.getInitiativeId());
+
+        Assertions.assertNull(expected.getNotificationDate());
+        Assertions.assertNotNull(actual.getNotificationDate());
+        Assertions.assertEquals(TODAY.plusDays(1), actual.getNotificationDate());
```

Comment:
you should also fetch and check the results querying the repo (for each result, search it into the repo and verify if they are equal